components/discord_bot/actions/list-users-with-emoji-reactions/list-users-with-emoji-reactions.mjs

```diff
@@ -0,0 +1,80 @@
+import flatten from "lodash.flatten";
+import uniqBy from "lodash.uniqby";
+import common from "../common.mjs";
+
+const { discord } = common.props;
+
+export default {
+  ...common,
+  key: "discord_bot-list-users-with-emoji-reactions",
+  name: "List Users that Reacted with Emoji",
+  description: "Return a list of users that reacted with a specified emoji.",
+  type: "action",
+  version: "0.0.1",
+  props: {
+    ...common.props,
+    emoji: {
+      propDefinition: [
+        discord,
+        "emoji",
+      ],
+    },
+    limit: {
+      default: 10,
+      propDefinition: [
+        discord,
+        "limit",
+      ],
+    },
+    after: {
+      propDefinition: [
+        discord,
+        "after",
+      ],
+    },
+  },
+  async run({ $ }) {
+    const {
+      emoji: decodedEmoji,
+      channelId,
+      after,
+      limit,
+    } = this;
+
+    const emoji = encodeURIComponent(`${decodedEmoji}`);
+
+    const messages = await this.paginateMessages({
+      $,
+      channelId,
+      max: 20,
+    });
+
+    const messageIds = messages.map((message) => message.id);
+
+    const requestUserReactions =
+      messageIds
+        .map((messageId) => {
+          return this.discord.getUserReactions({
+            $,
+            channelId,
+            emoji,
+            messageId,
+            after,
+            limit,
+          });
+        });
```

Comment:
Since `getUserReactions` uses `after` and `limit`, we should include pagination here as well.

Reply:
There is a question here I neeed to answer first, would it be better looping over messages or just asking the user for the exact `messageId` with a `prop`? I would rather go with `messageId`. What do you think about it?

Reply:
Integromat's version of this action asks for a `messageId`, so I agree that asking for a `messageId` is best.